### PR TITLE
[Config] Enable libxsmm by default for AVX cpu

### PR DESCRIPTION
**Pull request by daniil-sizov**
## Description
Enable libxsmm by default for AVX cpu. Improves performance for AVX by leveraging libxsmm

## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [x] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [x] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

## Changes
- [x] Removed unused xbyak code (breaking build before the change)

### CODE REVIEW

File: src/array/cpu/spmm.h [hunk 1]
```diff
@@ -19,56 +19,14 @@
 
 #include "spmm_binary_ops.h"
 #if !defined(_WIN32)
-#ifdef USE_AVX
```
Comment (BarclayII) [inline src/array/cpu/spmm.h:22]: Why is `USE_AVX` removed?

Comment (daniil-sizov) [reply src/array/cpu/spmm.h:22]: Interesting. Is there a reason it was set up this way? I initially thought `USE_AVX` was added to workaround some build failures.
Right now it seems redundant, it wraps the same code USE_LIBXSMM wraps

**Comment by BarclayII** [comment]
@dgl-bot

**Comment by daniil-sizov** [comment]
Hi @BarclayII, could you please retrigger the build bot?

**Comment by BarclayII** [comment]
@dgl-bot

**Comment by BarclayII** [comment]
@dgl-bot

**Comment by BarclayII** [comment]
@dgl-bot

**Review by BarclayII — APPROVED** [review]

**Pull request merged**